### PR TITLE
Add chromeos.kernelci.org script and config

### PR DESCRIPTION
**Pull request by mgalka**
Added chromeos.kernelci.org script based on staging.kernelci.org and
chromeos.ini file that keeps config specific to chromeos.kernelci.org

### CODE REVIEW

File: chromeos.kernelci.org [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        docker-compose down
+        docker-compose up --build -d
+        sleep 60  # magic - another way to do this would be to poll something
+    else
+        cd "$topdir/checkout/kernelci-jenkins"
+        # ToDo: restart the container but only when there were some changes in
+        # kernelci-jenkins to avoid aborting running jobs unnecessarily
+    fi
+    cd "$topdir"
+
```
Comment (gctucker) [inline chromeos.kernelci.org:48]: Please drop this, it's the same Jenkins instance as staging.kernelci.org.

File: chromeos.kernelci.org [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+cmd_backend() {
+    opt="$1"
+
+    echo "Checking for missing indexes by finding COLLSCAN events"
+    echo "db.system.profile.aggregate([{\$match: {ns: {\$in: ['chromeos-kernel-ci.test_case', 'chromeos-kernel-ci.test_group', 'chromeos-kernel-ci.test_regression', 'chromeos-kernel-ci.build']}, planSummary: 'COLLSCAN'}}, {\$group: {_id: {ns: '\$ns', planSummary: '\$planSummary'}}}])" | mongo chromeos-kernel-ci
+
+    if [ "$opt" = "indexes" ]; then
+        echo "Dropping Mongo DB indexes"
+        echo "db.test_case.dropIndexes(); db.test_group.dropIndexes(); db.test_regression.dropIndexes(); db.build.dropIndexes();" | mongo chromeos-kernel-ci
+    fi
```
Comment (gctucker) [inline chromeos.kernelci.org:82]: Please drop this, we might only enable it if we start making changes to the schema used for chromeos.kernelci.org.

File: chromeos.kernelci.org [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+    echo "Updating kernelci-backend"
+    ./pending.py kernelci-backend --settings=${CHROMEOS_SETTINGS} --main=chromeos.kernelci.org --push
+    # ToDo: check if there were any changes to skip if there weren't any
+    ./ansible kernelci-backend api.chromeos.kernelci.org
+
+    echo "Resetting and enabling the Mongo DB profiler"
+    echo "db.setProfilingLevel(0); db.system.profile.drop(); db.setProfilingLevel(2);" | mongo chromeos-kernel-ci
```
Comment (gctucker) [inline chromeos.kernelci.org:90]: Ditto

File: chromeos.kernelci.org [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    for d in external/*; do
+        echo "Checking out staging branch for $d"
+        cd "$d"
+        git remote update origin
+        git checkout origin/staging.kernelci.org
+        cd -
+    done
+    ./docker-hugo -D -b https://chromeos.kernelci.org:8080
+    cd "$topd"
+}
```
Comment (gctucker) [inline chromeos.kernelci.org:116]: Please drop this, there's no static website for chromeos.kernelci.org.

File: chromeos.kernelci.org [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+            ;;
+        stable)
+            url=https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git
+            branch=linux-5.4.y
+            ;;
+        *)
+            echo "Unknown tree"
+            exit 1
+            ;;
+    esac
```
Comment (gctucker) [inline chromeos.kernelci.org:140]: As discussed, please drop this as we're starting with only one kernel branch (stable 5.12) on chromeos.kernelci.org.

File: chromeos.kernelci.org [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+        --tag-prefix=chromeos-"$tree"-
+}
+
+cmd_docker() {
+    echo "Updating Docker images"
+    # ToDo: check if there were any changes to skip if there weren't any
+    cd checkout/kernelci-core/config/docker
+    ./build-and-push.sh -pbdikQ -t kernelci/staging-
+    cd -
+}
```
Comment (gctucker) [inline chromeos.kernelci.org:157]: Please drop this, the chromeos.kernelci.org instance uses the same Docker images as staging.

File: chromeos.kernelci.org [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+            tree=next
+            ;;
+        *)
+            tree=next
+            ;;
+    esac
+
+    echo $tree > .tree
+    echo $tree
+}
```
Comment (gctucker) [inline chromeos.kernelci.org:192]: Please drop this as well...

File: data/chromeos.ini [hunk 8]
```diff
@@ -0,0 +1,59 @@
+[DEFAULT]
+branch: chromeos.kernelci.org
+namespace: kernelci
+users:
```
Comment (gctucker) [inline data/chromeos.ini:4]: We might want to trim this down to the list of users who are actually going to be working on the chromeos.kernelci.org instance.

**Review by gctucker — CHANGES_REQUESTED** [review]
It would have been better to start with an empty file and add commands gradually rather than copying staging.kernelci.org and hacking it until it works.  There's a good chance we'll have unnoticed bugs while doing that.

**Comment by mgalka** [comment]
> It would have been better to start with an empty file and add commands gradually rather than copying staging.kernelci.org and hacking it until it works. There's a good chance we'll have unnoticed bugs while doing that.

Am I expected to write this script from scratch? I thought that writing a script which does pretty much the same as the one we already have doesn't make much sense. I deliberately marked this PR as draft and commented out all commands, but first two to run it several times and gradually enable subsequent steps.

File: data/chromeos.ini (hunk 8 above)
Comment (mgalka) [reply data/chromeos.ini:4]: I wanted to clarify that during tonight's Open Hours. I'll clear the users list.

File: chromeos.kernelci.org (hunk 1 above)
Comment (mgalka) [reply chromeos.kernelci.org:48]: I think that issue description should be changed then. At the moment it says that there should be a `chromeos.kernelci.org` branch created using pending PRs also for `kernelci-jenkins`

Comment (gctucker) [reply chromeos.kernelci.org:48]: We need to create the branch with pending PRs but no need to recreate the Docker container.  The jobs will be picking up the branch created by the script.

Comment (mgalka) [reply chromeos.kernelci.org:48]: Got it.

File: chromeos.kernelci.org [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+cmd_backend() {
+    echo "Updating kernelci-backend"
+    ./pending.py kernelci-backend --settings=${CHROMEOS_SETTINGS} --main=chromeos.kernelci.org --push
+    # ToDo: check if there were any changes to skip if there weren't any
+    ./ansible kernelci-backend api.chromeos.kernelci.org
+
+    echo "Resetting and enabling the Mongo DB profiler"
+    echo "db.setProfilingLevel(0); db.system.profile.drop(); db.setProfilingLevel(2);" | mongo chromeos-kernel-ci
```
Comment (gctucker) [inline chromeos.kernelci.org:38]: I still think we can drop this part, about Mongo DB profiler.  It's only really useful when changing the schema and adjusting indexes to ensure Mongo doesn't do full collection scans.

Comment (mgalka) [reply chromeos.kernelci.org]: done

File: data/chromeos.ini (hunk 8 above)
Comment (mgalka) [reply data/chromeos.ini:4]: I reduced the user list

**Comment by gctucker** [comment]
Replaced with #10

**Pull request closed without merging**